gef-ui/server/types/deal.ts

```diff
@@ -0,0 +1,23 @@
+import { DealStatus, DealType, IsoDateTimeStamp, AnyObject } from '@ukef/dtfs2-common';
+
+/**
+ * Deal object received from portal-api
+ *
+ * This type is likely incomplete and should be added
+ * to as and when new properties are discovered
+ */
+export type Deal = {
+  dealType: DealType;
+  version?: number;
+  maker: AnyObject;
+  status: DealStatus;
+  bank: AnyObject;
+  exporter: AnyObject;
+  bankInternalRefName: string | null;
+  createdAt: IsoDateTimeStamp;
+  updatedAt: IsoDateTimeStamp;
+  submissionCount: number;
+  supportingInformation: AnyObject;
+  editedBy: string[];
+  additionalRefName: string | null;
+};
```

Comment:
Could `bankInternalRefName` and `additionalRefName be the following?:

```ts
bankInternalRefName?: string;
additionalRefName?: string;
```

Reply:
`?` means it can be `undefined`. But these are always `string | null` (based on the `Facility` class in portal-api)

Reply:
Got it, thanks. Not for this PR, but ideally could look into updating the `Facility` class to return undefined instead of null - could simplify some things, but maybe unintended consequences from this, one for the future.